### PR TITLE
fix(common): support numeric value as discrete cases for NgPlural

### LINKED ISSUE #11578: [i18n][feature] use numeric value as discrete cases for NgPlural
Current usage

``` html
<some-element [ngPlural]="value">
  <ng-container *ngPluralCase="'=0'">there is nothing</ng-container>
  <ng-container *ngPluralCase="'=1'">there is one</ng-container>
  <ng-container *ngPluralCase="'few'">there are a few</ng-container>
  <ng-container *ngPluralCase="'other'">there are exactly #</ng-container>
</some-element>
```

Also allow numeric values (ie 0, 1 below):

``` html
<some-element [ngPlural]="value">
  <ng-container *ngPluralCase="0">there is nothing</ng-container>
  <ng-container *ngPluralCase="1">there is one</ng-container>
  <ng-container *ngPluralCase="'few'">there are a few</ng-container>
  <ng-container *ngPluralCase="'other'">there are exactly #</ng-container>
</some-element>
```

### PR DESCRIPTION
**Pull request by marclaval**
Fixes #11578

### CODE REVIEW

File: modules/@angular/common/src/directives/ng_plural.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ export class NgPluralCase {
   constructor(
       @Attribute('ngPluralCase') public value: string, template: TemplateRef<Object>,
       viewContainer: ViewContainerRef, @Host() ngPlural: NgPlural) {
-    ngPlural.addCase(value, new SwitchView(viewContainer, template));
+    ngPlural.addCase(
+        isNaN(Number(value)) ? value : `=${value}`, new SwitchView(viewContainer, template));
```
Comment (vicb) [inline modules/@angular/common/src/directives/ng_plural.ts]: - use [Number.isNaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) ?
- `const isANumber = ...;` (readabilty)

Comment (mlaval) [inline modules/@angular/common/src/directives/ng_plural.ts]: Done

File: modules/@angular/common/src/directives/ng_plural.ts [hunk 2]
```diff
@@ -103,6 +103,7 @@ export class NgPluralCase {
   constructor(
       @Attribute('ngPluralCase') public value: string, template: TemplateRef<Object>,
       viewContainer: ViewContainerRef, @Host() ngPlural: NgPlural) {
-    ngPlural.addCase(value, new SwitchView(viewContainer, template));
+    const isANumber: boolean = !(<any>Number).isNaN(Number(value));
```
Comment (vicb) [inline modules/@angular/common/src/directives/ng_plural.ts]: why `(<any>Number)` and not Number ?

might be that we don't allow this ES6 construct ?

Comment (DzmitryShylovich) [inline modules/@angular/common/src/directives/ng_plural.ts]: yeah, `Object.assign` is not available, for example

**Pull request closed without merging**